src/node/rpc/node_frontend.h

```diff
@@ -1515,6 +1516,10 @@ namespace ccf
           in.public_key};
         g.add_node(in.node_id, node_info);
         g.trust_node_code_id(in.code_digest, in.quote_info.format);
+        if (in.quote_info.format == QuoteFormat::amd_sev_snp_v1){
+          auto digest = EnclaveAttestationProvider::get_security_policy_digest(in.quote_info).value();
```

Comment:
Note that it matters much for performance, but you could simply re-hash `in.security_policy` instead of re-extracting from the quote here.

Reply:
Yeah I was hoping to have the refactor done where we would just read the appropriate field rather than re-verifying, I'll change this

Reply:
Oh wait, we can't do this if we want to allow user to not set the raw security policy